### ISSUE TITLE
Error when using password with semicolon

### ISSUE THREAD

**Issue by grungondola** [NLog/NLog#3548]
**NLog version**: 4.6.6

**Platform**: .Net 4.6.1 /  .NET Standard 2.0 (dual framework)

**Current NLog config**

```c#
// C# (Replaced some private/sensitive data with generic data)
LoggingConfiguration config = new LoggingConfiguration();
StringBuilder dbCommandBuilder = new StringBuilder();
dbCommandBuilder.AppendLine("INSERT INTO [dbo].[Log] ([ColumnA],[ColumnB],[ColumnC])");
dbCommandBuilder.Append("VALUES(@column_a, @column_b, @column_c)");
DatabaseTarget databaseTarget = new DatabaseTarget("database")
{
    DBDatabase = "DatabaseName",
    DBHost = "123.123.123.123",
    DBUserName = "Username",
    DBPassword = "passw;ord",
    CommandText = dbCommandBuilder.ToString(),
    DBProvider = "System.Data.SqlClient",
    KeepConnection = false
};
databaseTarget.Parameters.Add(new DatabaseParameterInfo("@exception", Layout.FromString("${exception:format=Message:innerFormat=Message:maxInnerExceptionLevel=10}"))); // Multiple of these for various parameters
config.AddTarget(databaseTarget);
config.AddRule(LogLevel.Warn, LogLevel.Fatal, databaseTarget);
LogManager.Configuration = config;

InternalLogger.LogToConsole = true;
InternalLogger.LogLevel = LogLevel.Error;
```

- What is the current result?
Not able to connect to the database because the password has a semicolon in it, which is being interpreted as the end of the password and the start of a new keyword
- What is the expected result?
I expect it to connect to the database using the password and log the errors (works in other environments with different passwords using the same code)
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
Yes
- Please post full exception details (message, stacktrace, inner exceptions)
```
2019-08-16 12:11:03.0113 Error DatabaseTarget(Name=database): Error when writing to database. Exception: System.ArgumentException: Keyword not supported: 'ord;database'.
at System.Data.Common.DbConnectionOptions.ParseInternal(Dictionary`2 parsetable, String connectionString, Boolean buildChain, Dictionary`2 synonyms, Boolean firstKey)
at System.Data.Common.DbConnectionOptions..ctor(String connectionString, Dictionary`2 synonyms)
at System.Data.SqlClient.SqlConnectionString..ctor(String connectionString)
at System.Data.SqlClient.SqlConnectionFactory.CreateConnectionOptions(String connectionString, DbConnectionOptions previous)
at System.Data.ProviderBase.DbConnectionFactory.GetConnectionPoolGroup(DbConnectionPoolKey key, DbConnectionPoolGroupOptions poolOptions, DbConnectionOptions& userConnectionOptions)
at System.Data.SqlClient.SqlConnection.ConnectionString_Set(DbConnectionPoolKey key)
at System.Data.SqlClient.SqlConnection.set_ConnectionString(String value)
at NLog.Targets.DatabaseTarget.OpenConnection(String connectionString)
at NLog.Targets.DatabaseTarget.EnsureConnectionOpen(String connectionString)
at NLog.Targets.DatabaseTarget.WriteEventToDatabase(LogEventInfo logEvent, String connectionString)
at NLog.Targets.DatabaseTarget.Write(LogEventInfo logEvent)
```
- Are there any workarounds? yes/no
Yes. Instead of specifying the DBHost, UserName, Password, and Database parameters separately, I can specify it as a connection string using single quotes around the password, and it works just fine.
- Is there a version in which it did work? 
Unsure
- Can you help us by writing an unit test?
Unsure

**Comment by 304NotModified** [comment]
checking this!

**Comment by 304NotModified** [comment]
@grungondola could you please test with 


```c#
DBPassword = "'passw;ord'",
```

We are building the connectionstring from the properties DBHost . DBDatabase , DBPassword & DBUserName ,


```
DBHost = "localhost",
DBPassword = "passw;ord",
DBUserName = "user",
DBDatabase = "MyDatabase"

```
will be now:
```
Server=localhost;User id=user;Password=passw;ord;Database=MyDatabase
```


and will result in

```
System.ArgumentException
Keyword not supported: 'ord;database'.
   at System.Data.Common.DbConnectionOptions.ParseInternal(Hashtable parsetable, String connectionString, Boolean buildChain, Hashtable synonyms, Boolean firstKey)
   at System.Data.Common.DbConnectionOptions..ctor(String connectionString, Hashtable synonyms, Boolean useOdbcRules)
   at System.Data.SqlClient.SqlConnectionString..ctor(String connectionString)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnectionOptions(String connectionString, DbConnectionOptions previous)
   at System.Data.ProviderBase.DbConnectionFactory.GetConnectionPoolGroup(DbConnectionPoolKey key, DbConnectionPoolGroupOptions poolOptions, DbConnectionOptions& userConnectionOptions)
   at System.Data.SqlClient.SqlConnection.ConnectionString_Set(DbConnectionPoolKey key)
   at System.Data.SqlClient.SqlConnection.set_ConnectionString(String value)
   at NLog.Targets.DatabaseTarget.OpenConnection(String connectionString) in D:\nlog\nlog\src\NLog\Targets\DatabaseTarget.cs:line 339
   at NLog.Targets.DatabaseTarget.EnsureConnectionOpen(String connectionString) in D:\nlog\nlog\src\NLog\Targets\DatabaseTarget.cs:line 751
   at NLog.Targets.DatabaseTarget.WriteEventToDatabase(LogEventInfo logEvent, String connectionString) in D:\nlog\nlog\src\NLog\Targets\DatabaseTarget.cs:line 661
   at NLog.Targets.DatabaseTarget.Write(LogEventInfo logEvent) in D:\nlog\nlog\src\NLog\Targets\DatabaseTarget.cs:line 566
   at NLog.Targets.Target.Write(AsyncLogEventInfo logEvent) in D:\nlog\nlog\src\NLog\Targets\Target.cs:line 574
   at NLog.Targets.Target.WriteAsyncThreadSafe(AsyncLogEventInfo logEvent) in D:\nlog\nlog\src\NLog\Targets\Target.cs:line 598
   at NLog.Targets.Target.WriteAsyncLogEvent(AsyncLogEventInfo logEvent) in D:\nlog\nlog\src\NLog\Targets\Target.cs:line 317
   at NLog.LoggerImpl.WriteToTargetWithFilterChain(Target target, FilterResult result, LogEventInfo logEvent, AsyncContinuation onException) in D:\nlog\nlog\src\NLog\LoggerImpl.cs:line 218
   at NLog.LoggerImpl.Write(Type loggerType, TargetWithFilterChain targetsForLevel, LogEventInfo logEvent, LogFactory factory) in D:\nlog\nlog\src\NLog\LoggerImpl.cs:line 99
   at NLog.Logger.WriteToTargets(LogEventInfo logEvent, TargetWithFilterChain targetsForLevel) in D:\nlog\nlog\src\NLog\Logger.cs:line 682
   at NLog.Logger.WriteToTargets(LogLevel level, IFormatProvider formatProvider, String message) in 

```


See also https://stackoverflow.com/questions/23014180/sql-server-2008-password-ending-in-a-semicolon

**Comment by 304NotModified** [comment]
Found the bug, fix is in progress

**Comment by snakefoot** [comment]
Resolved with #3566
